### ISSUE TITLE
Battery Meter stuck at 100%

### ISSUE THREAD

**Issue by SethDusek** [linuxmint/Cinnamon#4925]
Hi, I recently upgraded to mint 17.3, and hence cinnamon 2.8. But I noticed the battery meter is not changing at all from 100% while disconnected, even though acpi reports my battery at being 62% as you can see [here](http://i.shibe.ml/MPYQ.png)

**Comment by cromerc** [comment]
I am having the same issue. However if I plug in the power adapter the percentage updates, or if I unplug it, it updates. But it does not update on its own, it only updates when I plug or unplug the adapter.

**Comment by HileV** [comment]
I am having the same issue.

**Comment by keitalbame** [comment]
Same with me on Arch.

**Comment by HileV** [comment]
keitaibame, I think it's related to update systemd

**Comment by cromerc** [comment]
I am using Arch Linux as well. But I am not using systemd, I have OpenRC. So I don't think it's systemd that is the problem.

**Comment by HileV** [comment]
but.... why this applet didn't work?

**Comment by cromerc** [comment]
No idea, acpi shows the correct percentage and I have acpid running. I have upower installed as well. I installed cbatticon and it shows the correct percentage. It's just this applet that doesn't not work, everything else is working fine.

**Comment by HileV** [comment]
> cbatticon

Install just "cbatticon" or "cbatticon-gtk3" (gtk2) ?

**Comment by cromerc** [comment]
Cinnamon uses gtk3.

**Comment by HileV** [comment]
It needed to start on console? or restart xorg?

**Comment by cromerc** [comment]
You can put it in startup applications.

**Comment by HileV** [comment]
how to show percent on panel? For some reason the system tray no longer cause the context menu. It happened at the same time stopped working aplet battery.

**Comment by cromerc** [comment]
Sorry cbatticon does not show the percent on the panel, I am only using it as a temporary icon until this applet can be fixed.

**Comment by HileV** [comment]
In any case ur save my work!!!! THANK U BROTHER!!!!

**Comment by keitalbame** [comment]
Also noticed that the batery status on System Settings >> Power Management does not update automatically when you have the window open, to monitor.

Going back to System Settings and select Power Management again, does not update the values.
It only updates when the window is closed and open again.

After first installition of `acpi` and reboot on Arch (while on batteries), both applet and Power Managent updates the discharging rate but when I plugged in the AC adapter, the values were no longer updated.
Several reboots were made and values were no longer updated.
Removed acpi with `pacman -Rnu` and `pacman -Rns` options but after new installation, I was not able to reproduce:

> After first installition of `acpi` and reboot on Arch (while on batteries), both applet and Power Managent updates the discharging rate

After this point, no live updates are made without the plug/unplug "dance" for the applet, or reopening the Power Management window.

Both applet and Power Management correctly detects when charging or dischanging.


Didn't previously tested on Mint 17.3 but I tested now and it seems that is updating both applet and Power Managent on System Settings. 
I installed acpi just before I tested so don't know if it is related.
Plugging and unplugging AC adapter does not affect the values. They are always updated.

**Comment by HileV** [comment]
@keitalbame, on Mint applet is worked, but on Arch - dont work. WTF...

**Comment by HileV** [comment]
@cromerc this problem is not fixed?

**Comment by cromerc** [comment]
No, the issue is still there.

**Comment by HileV** [comment]
Damn.... Hard to wait when this trouble's fix....

**Comment by f2404** [comment]
The issue happened to me as well.
Noticed one thing - looks like running `$ upower -d` in the terminal not only shows the up-to-date information on the battery, but also updates the applet. Can serve as a temporary solution.

**Comment by HileV** [comment]
@f2404 Игорь, а если добавить `upower -d` в автозапуск как процесс?

**Comment by cromerc** [comment]
I found a solution for users with arch linux, if you downgrad to upower 99.3 it works. The problem with upower 99.4.

**Comment by HileV** [comment]
@cromerc Where to get a version 99.3 ?

**Comment by f2404** [comment]
@HileV https://archive.archlinux.org/packages/u/upower/

**Comment by HileV** [comment]
@f2404 thx. 99.3.1 ?

**Comment by f2404** [comment]
Yep, try it.

**Comment by HileV** [comment]
That still doesn't work. The pack rolled. Strange that it didn't work.

**Comment by f2404** [comment]
Noticed another thing - upower 0.99.4 is actually crashing when running `upower -d`.
`upowerd[16258]: segfault at 20 ip 00007fd952bf7a90 sp 00007ffc44705b48 error 4 in libglib-2.0.so.0.4600.2[7fd952b73000+10c000]`

**Comment by cromerc** [comment]
Did you reboot after installing it?

Chris Cromer

**Comment by aenavcam** [comment]
Maybe this is related to the port of upower to GDBus?

https://bugs.freedesktop.org/show_bug.cgi?id=45792

**Comment by HileV** [comment]
Maybe.....

**Comment by marmistrz** [comment]
For me `upower -d` doesn't update the applet. Downgrading fixes

**Comment by aenavcam** [comment]
This is fixed in https://github.com/linuxmint/cinnamon-settings-daemon/pull/113 
For archlinux users the patch was integrated into cinnamon-settings-daemon 2.8.4-2

**Comment by HileV** [comment]
@aenavcam thx

**Comment by keitalbame** [comment]
Seems to be working, thanks.

[xxx@xxx ~]$ sudo pacman -Q upower cinnamon-settings-daemon 
upower 0.99.4-2
cinnamon-settings-daemon 2.8.4-3

**Comment by Sven-Dawitz** [comment]
Thats interesting. For me, its not working with same version.

$ pacman -Q upower cinnamon-settings-daemon
upower 0.99.4-2
cinnamon-settings-daemon 2.8.4-3

**Comment by f2404** [comment]
@Sven-Dawitz, is what the applet is showing not consistent with `upower -d` output?

**Comment by Sven-Dawitz** [comment]
@f2404 exactly. As described earlier. Applet only updates if charger is removed/connected or if I force an update by changing "Show percentage"/"Show time remaining" in the configure options.

upower always show the right percentage.

**Comment by Sven-Dawitz** [comment]
TL;DR: Everything works, had to reboot.

excuse me. after checking my "uptime", I figured, I might not have restarted my system yet, and some in-memory components might be outdated.

after a full system reboot, everything works as expected.

**Comment by Vahan86** [comment]
@SethDusek, does this issue still relevant to Cinnamon 3.4?

**Comment by cromerc** [comment]
@Vahan86

This was a problem in 2.8.4, it was fixed a long time ago. It has nothing to do with 3.4.
